Assets/HoloToolkit/Common/Scripts/Singleton.cs

```diff
@@ -62,11 +72,48 @@ public static bool IsInitialized
             }
         }
 
+        public static bool ConfirmInitialized()
```

Comment:
This method seems rather redundant to `AssertIsInitialized()` and `IsInitialized` property.

Reply:
It's more useful then either of them.

Reply:
> It's more useful then either of them.

Why? I'm not familiar with what `T access = Instance;` is doing.

Reply:
Only when you attempt to access the singleton does it attempt to initialize it.  AssertIsInitialized just throws an exception if it hasn't happened to load yet, even if it would once accessed.  This actually tries to Initialize it then tells you if there is one.   Similarly, IsInitialized doesn't tell you if it would be initialized once you try to use it.